FrEee/Scripts/AI/Vanguard/MinistryOfColonization.csx

```diff
@@ -0,0 +1,296 @@
+        {
+            var buildPlan = new ShipBuildPlan()
+            {
+                IsComplete = false,
+                RequestPlanId = plan.PlanId,
+                SentOrder = false,
+                Role = plan.Planet.ColonizationAbilityName == "Colonize Planet - Gas Giant" ? "Colonize Planet - Gas" : plan.Planet.ColonizationAbilityName, 
+                Priority = plan.Priority, 
+                ConstructionQueue = plan.SourcePlanet?.ConstructionQueue
+            };
+            buildPlan.AssignId();
+            plan.ShipBuildPlanId = buildPlan.PlanId;
+            buildPlan.RequestPlanId = plan.PlanId; 
+            PlanManager.CurrentPlans.ShipBuildPlans.Add(buildPlan); 
+        }
+    }
+
+    /// <summary>
+    /// Updates any plan that has not yet truly started. 
+    /// </summary>
+    public void UpdateUnstartedPlans()
+    {
+        if (updatedPlans)
+            return; 
+        foreach(var plan in PlanManager.CurrentPlans.ColonizationPlans.Where(x => x.AwaitingBuild && x.ShipBuildPlanId == null))
+        {
+            //someone else settled this world, so cancel the plan. 
+            if (plan.Planet.Colony != null)
+            {
+                plan.IsComplete = true;
+                if (plan.ShipBuildPlan != null)
+                {
+                    plan.ShipBuildPlan.IsComplete = true;
+                }
+                continue;
+            }
+
+
+            int distance = int.MaxValue;
+            Planet sourcePlant;
+            foreach (var colony in Empire.ColonizedPlanets.Where(x => x.ConstructionQueue.IsSpaceYardQueue))
+            {
+                var curdistance = Pathfinder.EstimateDistance(colony.Sector, plan.Planet.Sector, Empire);
+                if (curdistance < distance)
+                {
+                    plan.SourcePlanetId = colony.ID;
+                    distance = curdistance;
+                    sourcePlant = colony;
+                }
+            }
+            //Lost connection to this world? set it to min priority. 
+            if (distance == int.MaxValue)
+            {
+                plan.Priority = int.MinValue; 
+                continue; 
+            }
+
+            CalculateScore(plan); 
+        }
+        updatedPlans = true; 
+    }
+
+    /// <summary>
+    /// Calculates the priority of the plan. 
+    /// </summary>
+    /// <param name="colonyPlan"></param>
+    void CalculateScore(ColonizationPlan colonyPlan)
+    {       
+        var planet = colonyPlan.Planet;
+        colonyPlan.Priority = 100; //initial value; 
+
+
+        colonyPlan.Priority -= Pathfinder.EstimateDistance(colonyPlan.SourcePlanet.Sector, planet.Sector, Empire);
+        if (planet.StarSystem.HasAbility(Empire, "Spaceport"))
+            colonyPlan.Priority += 100;
+        if (planet.Atmosphere == Empire.PrimaryRace.NativeAtmosphere)
+            colonyPlan.Priority += 500;
```

Comment:
So breathable planets get prioritized over all other planets, even if the other planets are larger? That would work in stock where a tiny breathable is the same as a huge domed, but not for ringworlds/sphereworlds. Maybe instead you could set the priority based on the number of facility slots and/or population capacity of the planet?

Reply:
hmmmm... I will look into tweaking that. Might tweak that to have an increase of priority for each facility slot available.